br/pkg/lightning/backend/tidb/tidb.go

```diff
@@ -351,7 +351,20 @@ func EncodeRowForRecord(encTable table.Table, sqlMode mysql.SQLMode, row []types
 	}
 	resRow, err := enc.Encode(log.L(), row, 0, columnPermutation, "", 0)
 	if err != nil {
-		return fmt.Sprintf("/* ERROR: %s */", err)
+		// if encode can't succeed, fallback to record the raw input strings
+		var res strings.Builder
+		for i, d := range row {
+			if i > 0 {
+				res.WriteByte(',')
+			}
+			s, err := d.ToString()
+			if err != nil {
+				// ignore the field if there are any error
+				s = ""
+			}
```

Comment:
🤔.

The only case `ToString()` returns an error is when `d.Kind()` is unknown which shouldn't happen. I'd suggest using `types.DatumsToString()` directly here.